packages/wallet/backend/src/grant/service.ts

```diff
@@ -80,6 +80,32 @@ export class GrantService implements IGrantService {
       args.filter = { identifier: { in: identifiers } }
     }
 
-    return await this.rafikiAuthService.listGrantsWithPagination(args)
+    const grants = await this.rafikiAuthService.listGrantsWithPagination(args)
+    this.parseIntervals(grants.grants.edges.map((it) => it.node))
+    return grants
```

Comment:
Usage of the non-mutating function would be sth like:
(Not tested)
```suggestion
   const grants = await this.rafikiAuthService.listGrantsWithPagination(args)
  const parsedGrants = this.parseIntervals(grants.grants.edges.map((it) => it.node))
  return {
    ...grants,
    grants: {
      ...grants.grants,
      edges: grants.grants.edges.map((edge, index) => ({ ...edge, node: parsedGrants[index] }))
    }
  }
```

Reply:
It would be harder to read than the current one.
and it would have more time complexity because we have one more for loop here

Reply:
anything that does not mutate data will implicitly have 1 extra loop